tensorflow/compiler/xla/service/llvm_ir/fused_ir_emitter.cc

```diff
@@ -306,7 +307,8 @@ bool FusedIrEmitter::IsFusedIrEmitterInefficient(
 
   // Check that the code duplication has at most a factor of 15 (where 15 is an
   // arbitrary constant that seems to work).
-  return total > 15 * index_usage_count.size();
+  const int64 index_usage_count_size_times_15 = 15 * index_usage_count.size(); 
+  return total > index_usage_count_size_times_15;
```

Comment:
This has the potential of a dangerous overflow if `index_usage_count.size()` is near max, multiplying by 15 would result in a small unsigned which then get converted to int with no issue.

Reply:
Hmm, how can I improve this?

Reply:
You should check before multiplication if it results in overflow.

```
const int64 max_size = kint64max / 16; // faster than dividing by 15
```

https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/platform/types.h;l=51;drc=e0266dbf39deac09315b764524835299b513926c

And then compare this with `index_usage_count.size()` before multiplying.

Reply:
Actually, looking at the usage of the function, this overflow seems acceptable as the only way this would be dangerous (`total` small, `index_usage_count.size()` large) would be if we overflow on `total` above. Let's go on with this as it is